ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -247,10 +247,12 @@ func validateRayServiceSpec(rayService *rayv1.RayService) error {
 	if headSvc := rayService.Spec.RayClusterSpec.HeadGroupSpec.HeadService; headSvc != nil && headSvc.Name != "" {
 		return fmt.Errorf("spec.rayClusterConfig.headGroupSpec.headService.metadata.name should not be set")
 	}
-
-	if upgradeStrategy := rayService.Spec.UpgradeStrategy; upgradeStrategy != nil {
-		if *upgradeStrategy != rayv1.NewCluster && *upgradeStrategy != rayv1.None {
-			return fmt.Errorf("spec.UpgradeStrategy value %s is invalid, valid options are %s or %s", *upgradeStrategy, rayv1.NewCluster, rayv1.None)
+	if rayService.Spec.UpgradeStrategy == nil {
```

Comment:
Can we merge these two `if` statements into one? The current control flow can easily skip some checks we might add in the future. For example, if UpgradeStrategy is nil, the "check others" step will be skipped.

```go
if rayService.Spec.UpgradeStrategy == nil {
  return nil
}

// check upgradeType
// check others
```

Reply:
maybe leave this as a good first issue for new contributors.